### PR TITLE
feat(AMBER-638): Enable artworks to be filtered by visibility level

### PR DESCRIPTION
**Pull request by lidimayra**
Jordan and Addy (curatorial) reached out about the need for a visibility level filter on Rosalind, so it's easier for them to exclude artworks that must not be included on marketing collections

https://github.com/artsy/rosalind/assets/8002618/0020faf7-93e0-435b-a684-5bcb0dd4d53d

(but even if those end up being included by accident, Gravity is also prepared to only consider visible to public artworks - as per https://github.com/artsy/gravity/pull/17630)

cc @artsy/amber-devs

### CODE REVIEW

File: app/javascript/components/batch_update/components/App.js [hunk 1]
```diff
@@ -0,0 +1,640 @@
+/* eslint-disable @typescript-eslint/no-this-alias */
```
Comment (jpotts244) [inline app/javascript/components/batch_update/components/App.js:1]: Woah! did you write all this JS?
Or Is it automatically generated?

File: app/javascript/components/batch_update/helpers/elasticsearch.js [hunk 2]
```diff
@@ -179,6 +183,17 @@ const publishedMatcher = publishedFilter => {
   }
 }
 
+const listedMatcher = listedFilter => {
+  switch (listedFilter) {
```
Comment (jpotts244) [inline app/javascript/components/batch_update/helpers/elasticsearch.js:187]: 🚀

**Review by jpotts244 — APPROVED** [review]
LGTM!

File: app/javascript/components/batch_update/components/App.js (hunk 1 above)
Comment (lidimayra) [reply app/javascript/components/batch_update/components/App.js:1]: oh! yeah, everything on App.js was automagically generated (I don't even know at what point this happened 🤔 )

**Review by damassi — APPROVED** [review]
❇️

File: app/javascript/components/batch_update/components/App.js (hunk 1 above)
Comment (lidimayra) [reply app/javascript/components/batch_update/components/App.js:1]: oh I just found out! I have it generated on main when I run `yarn tsc` (so I think it makes sense to have it versioned?)

Comment (damassi) [reply app/javascript/components/batch_update/components/App.js:1]: Ahh i need to look closer at things. If its not already in the repo, i'd remove it. I think what happened is that tsc also compiled the files (where elsewhere at artsy we only use it as a typechecker). I imagine that's what you were trying to do? fwiw, `yarn tsc --noEmit`

We should prob delete this as well as the .map file.

Comment (lidimayra) [reply app/javascript/components/batch_update/components/App.js:1]: Thank you both for catching this! Opening a PR for removing it [here](https://github.com/artsy/rosalind/pull/790)

Comment (anandaroop) [reply app/javascript/components/batch_update/components/App.js:1]: Ah yeah, we [began](https://github.com/artsy/rosalind/pull/672) to add Typescript support in 2022  (to this ca. 2016 app!) when we thought we were going to do a round of work in it, but then our roadmap changed so it's possible there are some rough edges in the TS support here. Thanks for catching that 🙏🏽

Comment (jpotts244) [reply app/javascript/components/batch_update/components/App.js:1]: 🙏🏻  thanks all!

**Pull request merged**